# Tailored MPC params:

Vehicle:
  m: 240                                              # Car's mass
  inertia: 180                                        # Interial moment
  gravity: 9.81                                       # Gravity's acceleration
  Ar: 1                                               # Area of air contact
  rho: 1.255                                          # Air's density
  width: 1.5                                          # Car's width
  longue: 2.72                                        # Car's length
  Lf: 0.708                                           # Distance from CoG to front axis
  Lr: 0.822                                           # Distance from CoG to rear axis
  d_IMU: -0.318                                       # Distance from vectornav to CoG
  Rwheel: 0.2                                         # Effective wheel radius

Topics:
  State: /AS/C/state
  Commands: /AS/C/commands
  Planner: /AS/C/trajectory/partial
  Tro: /AS/C/trajectory/full
  Finish: /AS/C/flags/finish
  Debug:
    Time: /AS/C/mpc/debug/time
    ExitFlag: /AS/C/mpc/debug/exitflags
    CurvatureTopic: /AS/C/mpc/debug/curvature
    ProgressTopic: /AS/C/mpc/debug/progress
  Vis:
    PredictedSteering:  /vis/predictedPath/steering
    PredictedPath:      /vis/predictedPath/path
    PredictedHeading:   /vis/predictedPath/heading
    ActualPath:         /vis/actualPlanner/path

NLOP:
  N: 40                                               # Prediction horizon
  Nslacks: 2                                          # Number of slacks variables

Hz: 20                                                #[s] MPC frequency
rk4_t: 0.025                                          #[s] Runge Kutta integration time
Nthreads: 5                                           # Number of threads

nPlanning: 1900                                       # n points saved from the planner

minVelFinish: 0.1                                     # Minimum velocity to consider the car at finish state


# Aproximated values for Cm & p_long compatibility (for D=3000, lambda=5):
#     p_long | Cm
#       1.0  |  6708 N      
#       2.0  |  4743.4 N      
#       3.0  |  3873 N      
#       5.0  |  3000 N      
#       5.5  |  2860.4 N  (default)   
#       6.0  |  2738.6 N      
#       11.0 |  2022.6 N      
#       20.0 |  1500 N  